File: Leetcode/BinaryTreeInorderTraversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> preorderTraversal(TreeNode root) {
        ArrayList<Integer> answer = new ArrayList<>();
        
        class Helper{
            public void preorderHelper(TreeNode helperRoot){
                if (helperRoot != null){
                    answer.add(helperRoot.val);
                    this.preorderHelper(helperRoot.left);
                    this.preorderHelper(helperRoot.right);
                }
            }
        }
        
        new Helper().preorderHelper(root);
        return answer;
    }
}

File: Leetcode/BinaryTreeInorderTraversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def inorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        answer = []
        
        def inorderHelper(helperRoot: Optional[TreeNode]):
            if helperRoot:
                inorderHelper(helperRoot.left)
                answer.append(helperRoot.val)
                inorderHelper(helperRoot.right)
        
        inorderHelper(root)
        
        return answer
